test/jdk/java/awt/print/Dialog/DialogCopies.java

```diff
@@ -29,284 +29,76 @@
+        if (job.getPrintService() == null) {
+            System.out.println("Looks like printer is not configured. Please install printer and " +
+                    "re-run the test case.");
+            testResult = false;
+            countDownLatch.countDown();
             return;
         }
-        if (job.getCopies() == 1) {
-            Sysout.printInstructions(errorMessage);
-        }
-
-        if (job.getCopies() > 1) {
-            Sysout.printInstructions(successMessage);
-        }
+        checkNoOfCopies(job, job.printDialog());
     }
 
-    /*****************************************************
-     Standard Test Machinery Section
-     DO NOT modify anything in this section -- it's a
-     standard chunk of code which has all of the
-     synchronisation necessary for the test harness.
-     By keeping it the same in all tests, it is easier
-     to read and understand someone else's test, as
-     well as insuring that all tests behave correctly
-     with the test harness.
-     There is a section following this for test-defined
-     classes
-     ******************************************************/
-    private static boolean theTestPassed = false;
-    private static boolean testGeneratedInterrupt = false;
-    private static String failureMessage = "";
-
-    private static Thread mainThread = null;
-
-    private static int sleepTime = 300000;
-
-    public static void main(String args[]) throws InterruptedException {
-        mainThread = Thread.currentThread();
-        try {
-            init();
-        } catch (TestPassedException e) {
-            //The test passed, so just return from main and harness will interpret this return as a pass
-            return;
-        }
-
-        /*
-            At this point, neither test passed nor test failed has been
-            called -- either would have thrown an exception and ended the
-            test, so we know we have multiple threads.
-            Test involves other threads, so sleep and wait for them to
-            called pass() or fail()
-         */
-        try {
-            Thread.sleep(sleepTime);
-            //Timed out, so fail the test
-            throw new RuntimeException("Timed out after " + sleepTime / 1000 + " seconds");
-        } catch (InterruptedException e) {
-            if (!testGeneratedInterrupt) throw e;
-
-            //reset flag in case hit this code more than once for some reason (just safety)
-            testGeneratedInterrupt = false;
-            if (theTestPassed == false) {
-                throw new RuntimeException(failureMessage);
-            }
+    public static void checkNoOfCopies(PrinterJob job, boolean pdReturnValue) {
+        if (pdReturnValue) {
+            System.out.println("User has selected OK/Print button on the PrintDialog");
+        } else {
+            System.out.println("User has selected Cancel button on the PrintDialog");
         }
```

Comment:
I suggest treating *Cancel* button as a pass. It will handle the case where the user does not have a real printer but has virtual printers such as *Microsoft Print to PDF* or *Microsoft XPS Document Writer*. Neither allows setting *Copies* to anything but 1. If you increase the number of copies and click OK, an error message is displayed. Yet clicking OK with number of copies set 1 will fail the test.

To avoid such a situation, clicking *Cancel* should be treated a passed test or rather a skipped one.

Reply:
If user click the Cancel button the test is passed.